app/src/main/java/org/literacyapp/authentication/helper/DetectionHelper.java

```diff
@@ -172,8 +185,64 @@ public static synchronized int getScreenBrightnessMode(Context context){
         }
     }
 
-    public static synchronized void setScreenBrightnessAndMode(Context context, int screenBrightnessMode, int screenBrightness){
+    public static synchronized int getDisplayTemperatureNight(){
+        try {
+            String display_temperature_night = runAsRoot(new String[] {"settings --cm get system display_temperature_night"});
+            return Integer.valueOf(display_temperature_night);
+        } catch (IOException e) {
+            Log.e(DetectionHelper.class.getName(), null, e);
+            return DISPLAY_TEMPERATURE_NIGHT_DEFAULT;
+        } catch (InterruptedException e) {
+            Log.e(DetectionHelper.class.getName(), null, e);
+            return DISPLAY_TEMPERATURE_NIGHT_DEFAULT;
+        }
+    }
+
+    public static synchronized void setScreenBrightnessAndMode(Context context, int screenBrightnessMode, int screenBrightness, int displayTemperatureNight){
         Settings.System.putInt(context.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS_MODE, screenBrightnessMode);
+        Log.i(DetectionHelper.class.getName(), "setScreenBrightnessAndMode: SCREEN_BRIGHTNESS_MODE set to " + screenBrightnessMode);
         Settings.System.putInt(context.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS, screenBrightness);
+        Log.i(DetectionHelper.class.getName(), "setScreenBrightnessAndMode: SCREEN_BRIGHTNESS set to " + screenBrightness);
+        try {
+            runAsRoot(new String[] {"settings --cm put system display_temperature_night " + displayTemperatureNight});
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        Log.i(DetectionHelper.class.getName(), "setScreenBrightnessAndMode: display_temperature_night set to " + displayTemperatureNight);
+    }
+
+    public static String runAsRoot(String[] commands) throws IOException, InterruptedException {
```

Comment:
As `runAsRoot` is not necessarily coupled with the `DetectionHelper`, can we move this method a separate helper class, e.g. `RootHelper`, so that it can be reused in other contexts in the future?